assets/src/edit-story/components/inspector/prepublish/prepublishChecklistProvider.js

```diff
@@ -47,46 +47,87 @@ function PrepublishChecklistProvider({ children }) {
     return { ...story, pages };
   });
 
+  const isStoryLoaded = story?.pages?.length > 0;
+  const isChecklistEmpty = checkpointState === PPC_CHECKPOINT_STATE.NO_ISSUES;
+
   const [currentList, setCurrentList] = useState([]);
+  const [isChecklistReviewRequested, setIsChecklistReviewRequested] = useState(
+    false
+  );
 
   const handleRefreshList = useCallback(async () => {
     const pagesWithSize = story.pages.map((page) => ({
       ...page,
       pageSize,
     }));
 
-    // discern PPC types of messages based on checkpointState
-    let types;
-    switch (checkpointState) {
-      case PPC_CHECKPOINT_STATE.UNAVAILABLE:
-        types = [];
-        break;
-      default:
-        types = [
-          PRE_PUBLISH_MESSAGE_TYPES.GUIDANCE,
-          PRE_PUBLISH_MESSAGE_TYPES.ERROR,
-          PRE_PUBLISH_MESSAGE_TYPES.WARNING,
-        ];
-        break;
-    }
-
     setCurrentList(
-      await getPrepublishErrors({ ...story, pages: pagesWithSize }, { types })
+      await getPrepublishErrors({ ...story, pages: pagesWithSize })
     );
-  }, [story, pageSize, checkpointState]);
+  }, [story, pageSize]);
 
   const prevPages = usePrevious(story.pages);
   const prevPageSize = usePrevious(pageSize);
 
   const refreshOnInitialLoad = prevPages?.length === 0 && story.pages?.length;
   const refreshOnPageSizeChange = prevPageSize?.width !== pageSize?.width;
 
+  const highPriorityLength = useMemo(
+    () =>
+      isStoryLoaded &&
+      currentList.filter(
+        (current) => current.type === PRE_PUBLISH_MESSAGE_TYPES.ERROR
+      ).length,
+    [currentList, isStoryLoaded]
+  );
+
+  const recommendedLength = useMemo(
+    () =>
+      isStoryLoaded &&
+      currentList.filter((current) =>
+        current.type.indexOf([
+          PRE_PUBLISH_MESSAGE_TYPES.GUIDANCE,
+          PRE_PUBLISH_MESSAGE_TYPES.WARNING,
+        ])
+      ).length,
+    [currentList, isStoryLoaded]
+  );
+
+  // Review dialog should be seen when there are high priority items and first publish still hasn't happened.
+  const shouldReviewDialogBeSeen =
+    highPriorityLength > 0 && !isChecklistReviewRequested;
+
   useEffect(() => {
     if (refreshOnInitialLoad || refreshOnPageSizeChange) {
       handleRefreshList();
     }
   }, [handleRefreshList, refreshOnInitialLoad, refreshOnPageSizeChange]);
 
+  useEffect(() => {
+    if (story.status === 'publish') {
+      dispatch(PPC_CHECKPOINT_ACTION.ON_STORY_IS_PUBLISHED);
+    }
+  }, [story]);
+
+  useEffect(() => {
+    if (
+      checkpointState === PPC_CHECKPOINT_STATE.ALL &&
+      highPriorityLength === 0 &&
+      recommendedLength === 0 &&
+      isStoryLoaded
+    ) {
+      dispatch(PPC_CHECKPOINT_ACTION.ON_ALL_ISSUES_CLEARED);
+    }
+  }, [checkpointState, isStoryLoaded, highPriorityLength, recommendedLength]);
+
+  useEffect(() => {
+    if (checkpointState === PPC_CHECKPOINT_STATE.NO_ISSUES) {
+      if (highPriorityLength > 0 || recommendedLength > 0) {
```

Comment:
nit: should we combine these into one line? Maybe it's better the initial way for readability
```suggestion
    if (checkpointState === PPC_CHECKPOINT_STATE.NO_ISSUES && (highPriorityLength > 0 || recommendedLength > 0)) {
```

Reply:
either way :) i think i was debugging something and that's why it's nested. i'll clean itu p.